<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QWidget" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>498</width>
    <height>335</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BT Demo</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_11">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="lb_logo_left">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="lb_logo_right">
       <property name="text">
        <string/>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QStackedWidget" name="stackedWidget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex">
      <number>2</number>
     </property>
     <widget class="QWidget" name="page_main">
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QListView" name="listView">
         <property name="autoFillBackground">
          <bool>false</bool>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="verticalScrollBarPolicy">
          <enum>Qt::ScrollBarAsNeeded</enum>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarAsNeeded</enum>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::NoSelection</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QPushButton" name="bt_scan">
           <property name="text">
            <string>Scan Remote Devices</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="bt_listen_spp">
           <property name="text">
            <string>Listen SPP</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="bt_quit">
           <property name="text">
            <string>Quit</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="bt_shutdown">
           <property name="text">
            <string>Quit BT</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_pair">
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_7">
         <item>
          <spacer name="verticalSpacer_5">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="label_pair">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Ensure that the remote device is in pairing/scanning mode.
Pairing remote device...</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
           <property name="wordWrap">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="bt_pair_cancel">
           <property name="text">
            <string>Cancel</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="bt_pair_done">
           <property name="text">
            <string>Done</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer_6">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_device">
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>34</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="bt_hid">
         <property name="text">
          <string>HID Test</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="bt_a2dp">
         <property name="text">
          <string>A2DP Test</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="bt_sink">
         <property name="text">
          <string>A2DP Sink Test</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="bt_spp">
         <property name="text">
          <string>SPP Test</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QPushButton" name="bt_trust">
           <property name="text">
            <string>Trust Device</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="bt_unpair">
           <property name="text">
            <string>Unpair Remote Device</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="bt_back">
           <property name="text">
            <string>Back</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_spp">
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="SPPTest" name="spp" native="true"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_hid">
      <layout class="QVBoxLayout" name="verticalLayout_9">
       <item>
        <widget class="HIDTest" name="hid" native="true"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_progress">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="sizeConstraint">
        <enum>QLayout::SetDefaultConstraint</enum>
       </property>
      </layout>
     </widget>
     <widget class="QWidget" name="page_a2dp">
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="A2DPTest" name="a2dp" native="true"/>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_a2dp_sink">
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="SINKTest" name="a2dp_sink" native="true"/>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>A2DPTest</class>
   <extends>QWidget</extends>
   <header>A2DPTest.h</header>
   <container>1</container>
   <slots>
    <signal>testFinished()</signal>
    <signal>deviceReady(bool)</signal>
    <slot>showMainPage()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>HIDTest</class>
   <extends>QWidget</extends>
   <header>HIDTest.h</header>
   <container>1</container>
   <slots>
    <signal>testFinished()</signal>
    <signal>deviceReady(bool)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>SPPTest</class>
   <extends>QWidget</extends>
   <header>SPPTest.h</header>
   <container>1</container>
   <slots>
    <signal>testFinished()</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>SINKTest</class>
   <extends>QWidget</extends>
   <header>SINKTest.h</header>
   <container>1</container>
   <slots>
    <signal>deviceReady(bool)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../bt_demo.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>bt_quit</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>quit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>373</x>
     <y>295</y>
    </hint>
    <hint type="destinationlabel">
     <x>539</x>
     <y>464</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bt_back</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>showMainPage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>438</x>
     <y>256</y>
    </hint>
    <hint type="destinationlabel">
     <x>543</x>
     <y>371</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bt_unpair</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>unpairDevice()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>282</x>
     <y>256</y>
    </hint>
    <hint type="destinationlabel">
     <x>537</x>
     <y>340</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bt_a2dp</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>testA2DP()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>126</x>
     <y>195</y>
    </hint>
    <hint type="destinationlabel">
     <x>543</x>
     <y>298</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bt_hid</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>testHID()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>144</x>
     <y>158</y>
    </hint>
    <hint type="destinationlabel">
     <x>539</x>
     <y>233</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bt_pair_cancel</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>cancelPair()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>146</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel">
     <x>543</x>
     <y>261</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bt_pair_done</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>showMainPage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>146</x>
     <y>222</y>
    </hint>
    <hint type="destinationlabel">
     <x>544</x>
     <y>294</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bt_shutdown</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>quitBT()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>478</x>
     <y>295</y>
    </hint>
    <hint type="destinationlabel">
     <x>539</x>
     <y>503</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>listView</sender>
   <signal>clicked(QModelIndex)</signal>
   <receiver>MainWindow</receiver>
   <slot>showDevice(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>117</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>538</x>
     <y>225</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bt_spp</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>testSPP()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>111</x>
     <y>225</y>
    </hint>
    <hint type="destinationlabel">
     <x>6</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bt_listen_spp</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>listenSPP()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>216</x>
     <y>295</y>
    </hint>
    <hint type="destinationlabel">
     <x>-4</x>
     <y>441</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bt_trust</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>trustDevice()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>88</x>
     <y>256</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>350</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>a2dp</sender>
   <signal>testFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>showMainPage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>47</x>
     <y>60</y>
    </hint>
    <hint type="destinationlabel">
     <x>6</x>
     <y>83</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hid</sender>
   <signal>testFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>showMainPage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>56</x>
     <y>60</y>
    </hint>
    <hint type="destinationlabel">
     <x>8</x>
     <y>376</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spp</sender>
   <signal>testFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>showMainPage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>102</x>
     <y>60</y>
    </hint>
    <hint type="destinationlabel">
     <x>8</x>
     <y>236</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>a2dp</sender>
   <signal>deviceReady(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>readyToTest(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>60</y>
    </hint>
    <hint type="destinationlabel">
     <x>606</x>
     <y>418</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hid</sender>
   <signal>deviceReady(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>readyToTest(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>60</y>
    </hint>
    <hint type="destinationlabel">
     <x>605</x>
     <y>278</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bt_scan</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>scanDevices()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>128</x>
     <y>295</y>
    </hint>
    <hint type="destinationlabel">
     <x>538</x>
     <y>438</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bt_opp</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>testOPP()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>378</x>
     <y>98</y>
    </hint>
    <hint type="destinationlabel">
     <x>539</x>
     <y>168</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bt_sink</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>testSINK()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>237</x>
     <y>180</y>
    </hint>
    <hint type="destinationlabel">
     <x>7</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>a2dp_sink</sender>
   <signal>deviceReady(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>readyToTest(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>178</y>
    </hint>
    <hint type="destinationlabel">
     <x>248</x>
     <y>167</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>quit()</slot>
  <slot>showMainPage()</slot>
  <slot>unpairDevice()</slot>
  <slot>testA2DP()</slot>
  <slot>testHSP()</slot>
  <slot>testHID()</slot>
  <slot>testFTP()</slot>
  <slot>testOPP()</slot>
  <slot>cancelPair()</slot>
  <slot>quitBT()</slot>
  <slot>scanDevices()</slot>
  <slot>showDevice(QModelIndex)</slot>
  <slot>playA2DP()</slot>
  <slot>pauseA2DP()</slot>
  <slot>stopA2DP()</slot>
  <slot>doneHID()</slot>
  <slot>doneA2DP()</slot>
  <slot>testSPP()</slot>
  <slot>doneSPP()</slot>
  <slot>msgWrite()</slot>
  <slot>listenSPP()</slot>
  <slot>trustDevice()</slot>
  <slot>readyToTest(bool)</slot>
  <slot>testSINK()</slot>
 </slots>
</ui>
